File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        d={}
        for i in arr:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        list_freq=list(d.values())
        check_freq={}
        for i in list_freq:
            if i in check_freq:
                return False
            check_freq[i]=1
        return True   
            
        
        
        
        
        

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        // Count occurrences of each element
        for (int i : arr) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        HashSet<Integer> set = new HashSet<>();
        
        // Check if all occurrence counts are unique
        for (int count : map.values()) {
            if (!set.add(count)) {  // add() returns false if the set already contains the element
                return false;
            }
        }
        
        return true;
    }
}